discovery/data_sharing/v0.1/data_sharing.md

```diff
@@ -0,0 +1,297 @@
+`eventType`. It MAY contain the keys `maxBatchSize` and/or
+`threshold`. The keys are defined as follows:
+
+* `category`: One of `content`, `account`. This is the category of
+  objects the FASP is interested in.
+* `subscriptionType`: One of `lifecycle`, `trends`. `lifecycle`
+  means the FASP would like to be notified when new objects of the given
+  type are created, when objects are updated and when objects are
+  deleted. `trends` means that the FASP would like to be notified of
+  objects that had recent interactions. `trends` is only applicable if
+  the `category` given is `content`.
+* `maxBatchSize`: The maximum number of events the FASP would like to
+  receive in a single request. This is optional.
+* `threshold`: If `eventType` is `trending` then this object can be used to
+  further specify when the event should be reported. Valid keys for this
+  object are:
+  * `timeframe`: Number of minutes in which interactions should fire an
+    event, defaults to `15`
+  * `shares`: Number of shares in the given timeframe, defaults to `3`
+  * `likes`: Number of likes in the given timeframe, defaults to `3`
+  * `replies`: Number of replies in the given timeframe, defaults to `3`
+
+Example object:
+
+```json
+{
+  "category": "content",
+  "subscriptionType": "trends",
+  "maxBatchSize": 10,
+  "threshold": {
+    "timeframe": 15,
+    "shares": 3
+  }
+}
+```
+
+The fediverse server MUST validate the request. If it is invalid it MUST
+return an HTTP status code `422` (Unprocessable Content).
+
+If it is valid the response from the fediverse server MUST include an
+HTTP status code `201` (Created) and a JSON object including the `id` of
+the generated subscription. The latter can be used to unsubscribe later.
+
+Example response object:
+
+```json
+{
+  "subscriptionId": 3446
+}
+```
+
+To unsubscribe the FASP can make an HTTP `DELETE` call to the
+`/data_sharing/v0/event_subscriptions/:id` endpoint, where `:id` is replaced
+with the ID received when the original subscription was created.
+
+Example call:
+
+```http
+DELETE /data_sharing/v0/event_subscriptions/3446
+```
+
+The response MUST be an HTTP status code `200` (OK).
+
+### Requesting Historic Content / Backfilling (FASP => Fediverse Server)
+
+To request historic content to index a FASP can make an HTTP `POST` call
+to the `/data_sharing/v0/backfill_requests` endpoint.
+
+Example call:
+
+```http
+POST /data_sharing/v0/backfill_requests
+```
+
+The request body MUST contain a JSON object with the following keys:
+
+* `category`: One of `content`, `account`. This is the category
+  of objects the FASP is interested in.
+* `maxCount`: An integer representing the maximum number of objects the
+  FASP would like to receive.
+* `cursor`: Optional. An opaque token received from the fediverse server
+  in a previous request. MUST be used by the fediverse server to
+  determine results that have not already been sent to the FASP.
```

Comment:
Once again, how does the Fediverse server know where to send the results to?

Reply:
That is explained below.